ENH: add a count_nonzero method to numpy.ma. (#18637)
numpy/numpy: doc/release/upcoming_changes/18637.improvement.rst

``count_nonzero`` has been added to ``np.ma``
----------------------------------------------

> When ``np.count_nonzero`` is applied to a masked array, it considers masked zero value as a non-zero value. Now, ``np.ma.count_nonzero`` has been implemented to avoid this issue by separating the array from the mask. The function returns the number of non-zero unmasked values in a given array. To use it, call ``np.ma.count_nonzero(arr, options)``. 

Reviewer: Please keep the lines < 80 characters in length.

Author: on it!

Reviewer 2: Sorry to be late, I think the doc is misleading: ``np.count_nonzero`` applied to a masked array *does not* consider masked zero value as non-zero values, but rather does inappropriately consider masked non-zero values. I would rather say simply:

``np.count_nonzero`` applied to a masked array does not consider mask; ``np.ma.count_nonzero`` has been implemented to avoid this issue.

``count_nonzero`` has been added to ``np.ma``
----------------------------------------------

> When ``np.count_nonzero`` is applied to a masked array, it considers masked 
> zero value as a non-zero value. Now, ``np.ma.count_nonzero`` has been 
> implemented to avoid this issue by separating the array from the mask. The 
> function returns the number of non-zero unmasked values in a given array. 
> To use it, call ``np.ma.count_nonzero(arr, options)``.

Reviewer 3: This can be much shorter.
```suggestion
The new ``np.ma.count_nonzero`` returns the number of non-zero unmasked values in
a given array,
```
